composer/utils/object_store/mlflow_object_store.py

```diff
@@ -0,0 +1,392 @@
+            default experiment name, or the experiment name specified by the `MLFLOW_EXPERIMENT_NAME` environment
+            variable if one is set.
+
+            If `experiment_id` is provided and `run_id` is not, the MLFlowObjectStore will create a new run in the
+            provided experiment.
+
+            Providing a `run_id` without an `experiment_id` will raise an error.
+        multipart_upload_chunk_size(int, optional): The maximum size of single chunk in an MLFlow multipart upload.
+            The maximum number of chunks supported by MLFlow is 10,000, so the max file size that can
+            be uploaded is `10 000 * multipart_upload_chunk_size`. Defaults to 100MB for a max upload size of 1TB.
+    """
+
+    def __init__(self, path: str, multipart_upload_chunk_size: int = 100 * 1024 * 1024) -> None:
+        try:
+            import mlflow
+            from mlflow import MlflowClient
+        except ImportError as e:
+            raise MissingConditionalImportError('mlflow', conda_package='mlflow>=2.9.2,<3.0') from e
+
+        try:
+            from databricks.sdk import WorkspaceClient
+        except ImportError as e:
+            raise MissingConditionalImportError('databricks', conda_package='databricks-sdk>=0.15.0,<1.0') from e
+
+        tracking_uri = os.getenv('MLFLOW_TRACKING_URI', MLFLOW_DATABRICKS_TRACKING_URI)
+        if tracking_uri != MLFLOW_DATABRICKS_TRACKING_URI:
+            raise ValueError(
+                'MLFlowObjectStore currently only supports Databricks-hosted MLFlow tracking. '
+                f'Environment variable `MLFLOW_TRACKING_URI` is set to a non-Databricks URI {tracking_uri}. '
+                f'Please unset it or set the value to `{MLFLOW_DATABRICKS_TRACKING_URI}`.')
+
+        # Use the Databricks WorkspaceClient to check that credentials are set up correctly.
+        try:
+            WorkspaceClient()
+        except Exception as e:
+            raise ValueError(
+                f'Databricks SDK credentials not correctly setup. '
+                'Visit https://databricks-sdk-py.readthedocs.io/en/latest/authentication.html#databricks-native-authentication '
+                'to identify different ways to setup credentials.') from e
+
+        self._mlflow_client = MlflowClient(MLFLOW_DATABRICKS_TRACKING_URI)
+        mlflow.environment_variables.MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.set(multipart_upload_chunk_size)
+
+        experiment_id, run_id, _ = MLFlowObjectStore.parse_dbfs_path(path)
+        if experiment_id == PLACEHOLDER_EXPERIMENT_ID:
+            experiment_id = None
+        if run_id == PLACEHOLDER_RUN_ID:
+            run_id = None
+
+        # Construct the `experiment_id` and `run_id` depending on whether format placeholders were provided.
+        if not dist.is_initialized() or dist.get_global_rank() == 0:
+            experiment_id, run_id = self._init_run_info(experiment_id, run_id)
+
+        if dist.is_initialized():
```

Comment:
dont think you need the initialized check here either

Reply:
See note above